lib/filters/http.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+var preq = require('preq');
+var utils = require('../utils');
+
+module.exports = function(hyper, req, next, options) {
+    options = options || {};
+    var host = req.uri.constructor === String ? req.uri : req.uri.protoHost;
+    var match;
+    if (options.allow) {
+        match = options.allow.find(function(allow) {
```

Comment:
Testing those sequentially could be quite expensive. Could we compile this into a single regexp, possibly using https://github.com/gwicke/regexp-utils/blob/master/index.js#L72 ?